sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1240,7 +1240,14 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       case SqlBaseParser.IN =>
         invertIfNotDefined(In(e, ctx.expression.asScala.map(expression)))
       case SqlBaseParser.LIKE =>
-        invertIfNotDefined(Like(e, expression(ctx.pattern)))
+        val escapeOpt = Option(ctx.escapeChar).map(string).map { str =>
+          if (str.length > 1) {
+            throw new ParseException("Invalid escape string." +
+              "Escape string must be empty or one character.", ctx)
```

Comment:
Any test for this exception?

Reply:
OK. I will add tests in `ExpressionParserSuite`.